### ISSUE TITLE
SM ffmpeg bundle vs system

### ISSUE THREAD

**Issue by latot** [stepmania/stepmania#863]
Hi, here a weird thing, i know sm only works with some versions of ffmpeg, the sm 5.1 use ffmpeg-2.8 and in my system works fine with bundled ffmpeg, but i try with system ffmpeg upgrading to the same version, but i only get a static image screen in the video (the background images works fine).

i'm using gentoo 64

SM Log:
"mem", "(cache)", "/@mem/"
"dir", "/", "/rootfs/"
"dir", "/proc", "/proc/"
"dir", "/home/pipe/Documentos/git/stepmania/.", "/"
"dir", "/home/pipe/.stepmania-5.0/Announcers", "/Announcers/"
"dir", "/home/pipe/.stepmania-5.0/BGAnimations", "/BGAnimations/"
"dir", "/home/pipe/.stepmania-5.0/BackgroundEffects", "/BackgroundEffects/"
"dir", "/home/pipe/.stepmania-5.0/BackgroundTransitions", "/BackgroundTransitions/"
"dir", "/home/pipe/.stepmania-5.0/Cache", "/Cache/"
"dir", "/home/pipe/.stepmania-5.0/CDTitles", "/CDTitles/"
"dir", "/home/pipe/.stepmania-5.0/Characters", "/Characters/"
"dir", "/home/pipe/.stepmania-5.0/Courses", "/Courses/"
"dir", "/home/pipe/.stepmania-5.0/Logs", "/Logs/"
"dir", "/home/pipe/.stepmania-5.0/NoteSkins", "/NoteSkins/"
"dir", "/home/pipe/.stepmania-5.0/Packages", "/UserPackages/"
"dir", "/home/pipe/.stepmania-5.0/Save", "/Save/"
"dir", "/home/pipe/.stepmania-5.0/Screenshots", "/Screenshots/"
"dir", "/home/pipe/.stepmania-5.0/Songs", "/Songs/"
"dir", "/home/pipe/.stepmania-5.0/RandomMovies", "/RandomMovies/"
"dir", "/home/pipe/.stepmania-5.0/Themes", "/Themes/"
StepMania5.0-git-5e3768e
Compiled 20151011 @ 15:34:19 (build 5e3768e)
Log starting 2015-10-11 12:44:37
Loading window: gtk
OS: Linux ver 040107
Runtime library: glibc 2.20
Threads library: NPTL 2.20
Sound driver: pulse
Lights driver: SystemMessage
Lights driver: Export
"dir", "/", "/@mc1int/"
"timeout", "/@mc1int/", "/@mc1/"
"dir", "/", "/@mc2int/"
"timeout", "/@mc2int/", "/@mc2/"
Video renderers: 'opengl'
Display: :0 (screen 0)
X server vendor: The X.Org Foundation [1.16.4.0]
Server GLX vendor: SGI [1.4]
Client GLX vendor: Mesa Project and SGI [1.4]
Paletted textures disabled: GL_EXT_paletted_texture missing.
Direct rendering: yes
OGL Vendor: Intel Open Source Technology Center
OGL Renderer: Mesa DRI Intel(R) Ivybridge Mobile 
OGL Version: 3.0 Mesa 10.5.8
OGL Max texture size: 8192
OGL Texture units: 8
GLU Version: 1.3
OGL Extensions:
  GL_3DFX_texture_compression_FXT1
  GL_AMD: conservative_depth, draw_buffers_blend, performance_monitor, seamless_cubemap_per_texture, 
    shader_trinary_minmax
  GL_ANGLE: texture_compression_dxt3, texture_compression_dxt5
  GL_APPLE: object_purgeable, packed_pixels, vertex_array_object
  GL_ARB: ES2_compatibility, ES3_compatibility, blend_func_extended, buffer_storage, clear_buffer_object, 
    clear_texture, color_buffer_float, compressed_texture_pixel_storage, conditional_render_inverted, 
    conservative_depth, copy_buffer, copy_image, debug_output, depth_buffer_float, depth_clamp, depth_texture, 
    derivative_control, draw_buffers, draw_buffers_blend, draw_elements_base_vertex, draw_instanced, 
    explicit_attrib_location, explicit_uniform_location, fragment_coord_conventions, fragment_program, 
    fragment_program_shadow, fragment_shader, framebuffer_object, framebuffer_sRGB, get_program_binary, 
    half_float_pixel, half_float_vertex, instanced_arrays, internalformat_query, invalidate_subdata, 
    map_buffer_alignment, map_buffer_range, multi_bind, multisample, multitexture, occlusion_query, 
    occlusion_query2, pixel_buffer_object, point_parameters, point_sprite, provoking_vertex, robustness, 
    sample_shading, sampler_objects, seamless_cube_map, seamless_cubemap_per_texture, separate_shader_objects, 
    shader_atomic_counters, shader_bit_encoding, shader_objects, shader_texture_lod, shading_language_100, 
    shading_language_420pack, shading_language_packing, shadow, sync, texture_border_clamp, texture_compression, 
    texture_compression_bptc, texture_compression_rgtc, texture_cube_map, texture_cube_map_array, texture_env_add, 
    texture_env_combine, texture_env_crossbar, texture_env_dot3, texture_float, texture_gather, 
    texture_mirror_clamp_to_edge, texture_mirrored_repeat, texture_multisample, texture_non_power_of_two, 
    texture_query_levels, texture_query_lod, texture_rectangle, texture_rg, texture_rgb10_a2ui, texture_storage, 
    texture_storage_multisample, texture_swizzle, texture_view, timer_query, transform_feedback2, 
    transform_feedback3, transform_feedback_ins

![img](https://cloud.githubusercontent.com/assets/10523095/10417584/d868bca4-701f-11e5-86dc-2149ea8c2071.png)

**Comment by wolfman2000** [comment]
I wish to confirm the situation before I respond further.

Does the bundled version work? (2.1.3)

Does the upgraded system version work? (2.8.x)

It could've our code does not yet work for later versions.

**Comment by kyzentun** [comment]
Just to be clear:  We're not opposed to using a later version of ffmpeg in stepmania.  We just don't have anyone to go update the code in stepmania that uses ffmpeg.  If someone wants to step up and update src/arch/MovieTexture/MovieTexture_FFMpeg to a newer version of ffmpeg, that would be nice.

**Comment by latot** [comment]
okis, in the sm 5.1
bundled 2.1.3 works
bundled 2.1.8 works
bundled 2.8 works
system 2.8 don't works

in 5.1 sm use ffmpeg as submodule of git, actually the 2.8.

**Comment by wolfman2000** [comment]
Wait, what? We can actually try to bundle ffmpeg 2.8 in our code now?

If that is the case, then that is impressive. But then that also brings up a separate question of why the system ffmpeg fails.

Is there a way of confirming the system configure flags used?

**Comment by latot** [comment]
configure --prefix=/usr --libdir=/usr/lib32 --shlibdir=/usr/lib32 --mandir=/usr/share/man --enable-shared --cc=x86_64-pc-linux-gnu-gcc -m32 --cxx=x86_64-pc-linux-gnu-g++ -m32 --ar=ar --optflags=  --disable-static --enable-avfilter --enable-avresample --disable-stripping --enable-version3 --enable-version3 --enable-nonfree --disable-indev=v4l2 --disable-outdev=v4l2 --disable-indev=oss --disable-outdev=oss --enable-version3 --enable-nonfree --enable-bzlib --enable-runtime-cpudetect --disable-debug --enable-doc --disable-gnutls --enable-gpl --enable-hardcoded-tables --enable-iconv --enable-lzma --enable-network --disable-openssl --enable-postproc --enable-libsmbclient --enable-ffplay --enable-vaapi --enable-vdpau --enable-xlib --enable-libxcb --enable-libxcb-shm --enable-libxcb-xfixes --enable-zlib --enable-libcdio --enable-libiec61883 --enable-libdc1394 --disable-libcaca --enable-openal --enable-opengl --disable-libv4l2 --enable-libpulse --enable-libopencore-amrwb --enable-libopencore-amrnb --enable-libfdk-aac --enable-libopenjpeg --enable-libbluray --enable-libcelt --enable-libgme --enable-libgsm --enable-libmodplug --enable-libopus --enable-libquvi --disable-librtmp --disable-libssh --disable-libschroedinger --enable-libspeex --enable-libvorbis --enable-libvpx --enable-libzvbi --enable-libbs2b --enable-libflite --enable-frei0r --enable-libfribidi --enable-fontconfig --enable-ladspa --enable-libass --enable-libfreetype --enable-libsoxr --enable-pthreads --enable-libvo-aacenc --enable-libvo-amrwbenc --enable-libmp3lame --enable-libaacplus --enable-libfaac --disable-libsnappy --enable-libtheora --enable-libtwolame --enable-libwavpack --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxvid --disable-frei0r --disable-amd3dnow --disable-amd3dnowext --disable-avx2 --disable-fma3 --disable-fma4 --disable-xop --enable-pic --disable-asm --cpu=host

**Comment by wolfman2000** [comment]
That's...a lot of configuration options. Is there a way you can remove the ones that are applied by default?

**Comment by latot** [comment]
mm, i can't do it, in gentoo the system compile every program, so this conf it the passed to the compiler, but i can show this:

USE="X aac aacplus alsa amr amrenc bluray bs2b bzip2 cdio celt cpudetection doc encode faac fdk flite fontconfig frei0r fribidi gme gpl gsm hardcoded-tables iconv iec61883 ieee1394 jack jpeg2k ladspa libass libsoxr lzma modplug mp3 network openal opengl opus pic postproc pulseaudio quvi samba sdl speex theora threads truetype twolame vaapi vdpau vorbis vpx wavpack webp x264 x265 xcb xvid zlib zvbi (-altivec) (-armv5te) (-armv6) (-armv6t2) (-armvfp) -debug -examples -gnutls -libcaca -librtmp -libv4l (-mipsdspr1) (-mipsdspr2) (-mipsfpu) (-neon) -openssl -oss -schroedinger -snappy -ssh -static-libs {-test} -v4l"

every word represent a enabled option, the -word disabled option, the default options passed are this:
                --enable-avfilter --enable-avresample --disable-stripping

**Comment by wolfman2000** [comment]
...okay, guess I have to parse these manually. I'll report back in a day or two.

**Comment by latot** [comment]
before this, i'm little confusing, sm 5.1 use ffmpeg directly from git, without specify a branch, so i asume it use the 2.8 version, and because the place where the tags are selected was gone, but i want confirm this, how i can be sure the version of ffmpeg?

**Comment by latot** [comment]
well now i can confirm at least sm 5.0.x works fine with ffmpeg-2.8 bundle

**Comment by latot** [comment]
hi, following your idea i recompile sm with bundled ffmpeg-2.8 with the same flags of the system ffmpeg, and it works correctly.

**Comment by wolfman2000** [comment]
I'll address the comments in order.

The version of ffmpeg in use can be confirmed in the `.gitmodules` file (for `5_1_0`) or the SetupFfmpeg file (for `master`). Currently, they are tied to version 2.1.3.

Again, it sounds like 2.8 can be utilized.

As far as static vs shared goes, that is something that I may have to look into. Apparently shared is better for license compliance. I can look into that.

**Comment by latot** [comment]
mm, thx, in sm 5.0.x we can use setupffmpeg file to change the ffmpeg version and flags and works correctly, there i can use ffmpeg-2.8 bundle, but in sm 5.1 it uses ffmpeg as submodule, but the .gitmodules don't specify a branch, i can't find where specified the tag n2.1.3 to change it.

if you try to compile bundle 2.8 normally you need add -llzma to the linker.
if you want try with the big list of flags of my system you will need this flags to the linker:
-llzma -lvo-amrwbenc -lwebp -lwebpdecoder -lwebpdemux -lwebpmux -lopenjp2 -ltheora -ltheoraenc -ltheoradec -lfdk-aac -lgsm -lopencore-amrnb -lopencore-amrwb -lopenjpeg -lopus -lspeex -lwavpack -lcelt0 -lvorbisenc -lxvidcore -lvorbis -lbluray -lvpx -laacplus -lfaac -lgme -lmodplug -lmp3lame -lquvi -lsmbclient -ltheoraenc -ltwolame -lvo-aacenc -lx264 -lx265 -lzvbi -lvorbisfile -lswresample -lm -lz

anyway, the main issue continue, works ffmpeg2.8 bundle but don't works the same with system ffmpeg.

**Comment by wolfman2000** [comment]
https://github.com/FFmpeg/FFmpeg/releases  Tag and commit hash are located there. I will update the hash if #869 works out.

Speaking of that PR, it has been prepared. Windows side needs some adjusting before this can be merged. See [Appveyor](https://ci.appveyor.com/project/wolfman2000/stepmania/build/5.0.9.84.wolf-ffmpeg-2.8)

**Comment by latot** [comment]
hi, sorry continue this dude for mi ignorance, i don't know how sm 5.1 set ffmpeg to 2.1.3, in the .gitmodules say this:

[submodule "extern/ffmpeg-git"]
        path = extern/ffmpeg-git
        url = git://source.ffmpeg.org/ffmpeg.git

don't specify any tag or branch, but when we update the submodules say this:

Submodule path 'extern/ffmpeg-git': checked out 'eda6effcabcf9c238e4635eb058d72371336e09b'

basically the 2.1.3 tag, i try changing the ffmpeg version in setupffmpeg but no luky the extern/ffmpeg-git continues with 2.1.3.

**Comment by latot** [comment]
well, i close this and split it in two issues.

**Comment by lvazquez1** [comment]
Hi man, wich version of ffmpeg did u use with stempania 5 (linux)??? I used the git version, but some steps are delayed, so I wanna use sm5,but it crashes with videos, can u help me?

**Comment by latot** [comment]
open a new issue explining the situation, and include a backtrace, fast:
compile stepmania with debug symbols, in ccmake->build type write "Debug"
then open stepmania with gdb ```gdb stepmania``` in the compilation dir, then ```run```
crash the game, then click ```bt``` and paste the output

Cya.
